WordPressKit/RemoteVideoPressVideo.swift

```diff
@@ -0,0 +1,128 @@
+    ///
+    /// However, it's missing the following parameters that could be added in the future if needed:
+    /// - files
+    /// - file_url_base
+    /// - upload_date
+    /// - files_status
+    /// - subtitles
+    public var id: String
+    public var title: String?
+    public var videoDescription: String?
+    public var width: Int?
+    public var height: Int?
+    public var duration: Int?
+    public var displayEmbed: Bool?
+    public var allowDownload: Bool?
+    public var rating: String?
+    public var privacySetting: VideoPressPrivacySetting = .siteDefault
+    public var posterURL: URL?
+    public var originalURL: URL?
+    public var watermarkURL: URL?
+    public var bgColor: String?
+    public var blogId: Int?
+    public var postId: Int?
+    public var finished: Bool?
+
+    public var token: String?
+
+    enum CodingKeys: String, CodingKey {
+        case id, title, description, width, height, duration, displayEmbed, allowDownload, rating, privacySetting, posterURL, originalURL, watermarkURL, bgColor, blogId, postId, finished, token
+    }
+
+    public init(dictionary metadataDict: NSDictionary, id: String) {
+        self.id = id
+
+        title = metadataDict.string(forKey: "title")
+        videoDescription = metadataDict.string(forKey: "description")
+        width = metadataDict.number(forKey: "width")?.intValue
+        height = metadataDict.number(forKey: "height")?.intValue
+        duration = metadataDict.number(forKey: "duration")?.intValue
+        displayEmbed = metadataDict.object(forKey: "display_embed") as? Bool
+        allowDownload = metadataDict.object(forKey: "allow_download") as? Bool
+        rating = metadataDict.string(forKey: "rating")
+        if let privacySettingValue = metadataDict.number(forKey: "privacy_setting")?.intValue, let privacySettingEnum = VideoPressPrivacySetting.init(rawValue: privacySettingValue) {
+            privacySetting = privacySettingEnum
+        }
+        if let poster = metadataDict.string(forKey: "poster") {
+            posterURL = URL(string: poster)
+        }
+        if let original = metadataDict.string(forKey: "original") {
+            originalURL = URL(string: original)
+        }
+        if let watermark = metadataDict.string(forKey: "watermark") {
+            watermarkURL = URL(string: watermark)
+        }
+        bgColor = metadataDict.string(forKey: "bg_color")
+        blogId = metadataDict.number(forKey: "blog_id")?.intValue
+        postId = metadataDict.number(forKey: "post_id")?.intValue
+        finished = metadataDict.object(forKey: "finished") as? Bool
+    }
+
+    /// Returns the specified URL adding the token as a query parameter, which is required to play private videos.
+    /// - Parameters:
+    ///   - url: URL to include the token.
+    ///
+    /// - Returns: The specified URL with the token as a query parameter.
+    @objc(getURLWithToken:)
+    public func getURLWithToken(url: URL?) -> URL? {
```

Comment:
[According to the test](https://github.com/wordpress-mobile/WordPressKit-iOS/pull/581/files#diff-a12b9fab5b497cd6f553adf161c95b8cb62337a3fb0ae518fe5791d81948c4f3R58), we're expecting this method to return the original URL if a token doesn't exist.

**Optional suggestion:** Due to the name of the function and its return type, I'd be tempted to have this return a non-nil URL only if it can make a URL with the token. I'd also consider dealing with the optional URL early ([style guide ref](https://github.com/wordpress-mobile/swift-style-guide#deal-with-optionals-early)) and put the responsibility of this check on the caller.

Tests and callers would need to be updated, but the function could then be simplified to something like:

``` swift
    public func getURLWithToken(url: URL) -> URL? {
        guard
            let token,
            var urlComponents = URLComponents(url: url, resolvingAgainstBaseURL: true)
        else {
            return nil
        }

        let metadataTokenParam = URLQueryItem(name: "metadata_token", value: token)
        urlComponents.queryItems = (urlComponents.queryItems ?? []) + [metadataTokenParam]
        return urlComponents.url
    }
```

Reply:
The idea behind returning the URL when there's no token was to simplify the caller's code, but it's true that it can be misleading that the function might return the URL without the token if it's not defined. I'll update it with your suggestion, thanks!

Reply:
Suggestion applied in https://github.com/wordpress-mobile/WordPressKit-iOS/pull/581/commits/d77b9479c0c1a9e13493229f03ad67aa12ba540a.

Reply:
Heads up that due to the major version bump, we also need to bump the `WordPressAuthenticator` library: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/754.

Reply:
Good catch @fluiddot!